### PR TITLE
Replace Plasma Cutter EMP effect

### LINKED ISSUE #43685: Plasma Cutters are affected by EMP
They are recharged by plasma and not electricity. Seems silly that the loose all their charge from EMP

### PR DESCRIPTION
**Pull request by SkeletalElite**
## About The Pull Request
Remake of #44019

## Why It's Good For The Game
Fixes #43685

## Changelog

:cl:
add: new plasma cutter EMP effect, it sets you on fire
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/energy/special.dm [hunk 1]
```diff
@@ -167,6 +167,16 @@
 	else
 		..()
 
+/obj/item/gun/energy/plasmacutter/emp_act(severity)
+	if(!cell.charge)
+		return
+	cell.use(cell.charge/3)
+	if(isliving(loc))
+		var/mob/living/L = loc
```
Comment (JohnFulpWillard) [inline code/modules/projectiles/guns/energy/special.dm:175]: please no single letter variable :(

File: code/modules/projectiles/guns/energy/special.dm [hunk 2]
```diff
@@ -167,6 +167,16 @@
 	else
 		..()
 
+/obj/item/gun/energy/plasmacutter/emp_act(severity)
+	if(!cell.charge)
+		return
+	cell.use(cell.charge/3)
+	if(isliving(loc))
+		var/mob/living/user = loc
+		user.visible_message("<span class='danger'>Concentrated plasma discharges from [src] onto [user], burning them!</span>", "<span class='userdanger'>[src] malfunctions, spewing concentrated plasma onto you! It burns!</span>")
```
Comment (Fikou) [inline code/modules/projectiles/guns/energy/special.dm:176]: use span_danger()

Comment (Fikou) [reply code/modules/projectiles/guns/energy/special.dm:176]: and span user_danger()

**Comment by timothymtorres** [comment]
Very nice fix amigo.

**Pull request merged**